library/proc_macro/src/lib.rs

```diff
@@ -1074,7 +1074,11 @@ impl Literal {
         if !n.is_finite() {
             panic!("Invalid float literal {}", n);
         }
-        Literal(bridge::client::Literal::float(&n.to_string()))
+        let mut repr = n.to_string();
+        if !repr.contains('.') {
+            repr.push_str(".0");
```

Comment:
From reading the standard library code it looks like `to_string` should never produce an exponential form, so this shouldn't result in something like `1e100.0`.

The `Debug` impl for floats does exactly what we need (`10.0`) for small floats, but it does produce exponential forms for large floats.

Too bad `fn float_to_decimal_common_shortest` in `library\core\src\fmt\float.rs` is private, it can do exactly what we need here (although in theory it can be made public, but unstable and hidden, for `library/proc_macro`).

It's actually unfortunate that there's no public way to "print float as a float" in the standard library.
I wonder why the `Display` impl doesn't use the minimal precision 1 by default.

Reply:
I added a test involving `1e100` so that at least something breaks if `to_string` starts turning it into `1e100.0`, though I don't expect that change to ever get made.